cmd/main.go

```diff
@@ -96,6 +100,26 @@ func main() {
 	}
 }
 
+func setupMonitoringOrDie() {
+	if *metricsAddr == "" {
+		glog.Info("No metrics_listen address provided so skipping prometheus setup")
+		return
+	}
+
+	go func() {
+		http.Handle("/metrics", promhttp.Handler())
```

Comment:
I think this may expose the metrics on the `:8080` addr too (I think this adds the handler to the `http` package default servemux). It doesn't look like that's what you want so you may need to add your own `ServeMux` to your `srv` below instead and register `promhttp.Handler` on that.

Reply:
Fixed.